### DISCUSSION TITLE
LDAP Setup under Main Settings Menu

### DISCUSSION THREAD

**Discussion by Raven942** [kimai/kimai#3054] [General]
Under authentication, there are options to enable or disable certain authentication options and features. I would like to see LDAP integrated into that section so that it can be activated, configured, tested and tweaked. I noticed Snipe-IT has a feature similar to what I'm speaking about and it seems to be far more intuitive.

**Comment by kevinpapst** [comment]
> and it seems to be far more intuitive

Wow, I am so sorry to disappoint you that my free software is so complicated to use 😢 

And because you asked in such a friendly way, I tell you a secret. You can activate this code, test it and let me know what happens:
https://github.com/kevinpapst/kimai2/blob/1.16.10/src/Controller/SystemConfigurationController.php#L298

**Reply by Raven942** [reply]
No offense intended! your free is software is incredible. I was simply stating that changing the settings that way is easier for everyone (except you who has to make it work!)

**Reply by Raven942** [reply]
I noticed it gives us an activation checkbox, i'm assuming this is a sign of more to come? :)

**Reply by kevinpapst** [reply]
Remove `/*` in line 298 and `*/` in line 309

I have no clue what happens though. There will be new settings in the UI, but I can't remember if it does anything. 
I assume it works, but I do not show it yet, because all other settings need to be adjusted in the config file.

**Reply by Raven942** [reply]
yes, i posted prematurely, afterward I noticed the code break, and activated it. For your reference, I don't believe it does anything at this point and time. checking and unchecking did not seem to make a difference in login, then again, it may be broken on my side. I see that you have bigger plans in the future, which is great!

LDAP is something we need, however i've been struggling to set it up properly in Kimai

I appreciate your time on my humble request! Thank you for all that you do!

**Reply by kevinpapst** [reply]
All good 😁 

LDAP works for many clients and users, so if you share your config and the debug logs with errors, you might get help.

**Reply by Raven942** [reply]
link to my post in the appropriate place for follow up:

https://github.com/kevinpapst/kimai2/discussions/3056
